### PR TITLE
Refactored OpenAPIUIBundlesUpdater to use service listeners for OpenA…

### PR DESCRIPTION
**Pull request by KeenanMcCarthy**
…PI-3.1

Modified OpenAPIUIBundlesUpdater in com.ibm.ws.microprofile.openapi.impl.OpenAPIUIBundlesUpdater

### CODE REVIEW

**Comment by KeenanMcCarthy** [comment]
#libby
#build

File: dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/impl/OpenAPIUIBundlesListener.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
```
Comment (tjwatson) [inline dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/impl/OpenAPIUIBundlesListener.java]: The header needs to be updated to use EPL-2.0, also use copyright date of 2023 now.

```suggestion
/*******************************************************************************
 * Copyright (c) 2023 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-2.0/
 * 
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
```

**Review by tjwatson — COMMENTED** [review]
Looks like `com.ibm.ws.openapi31.customization.OpenAPIUIBundlesUpdater` also needs updated.

File: dev/com.ibm.ws.openapi.3.1/src/com/ibm/ws/openapi31/customization/OpenAPIUIBundlesListener.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
```
Comment (tjwatson) [inline dev/com.ibm.ws.openapi.3.1/src/com/ibm/ws/openapi31/customization/OpenAPIUIBundlesListener.java]: Need the EPL-2.0 header

File: dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/impl/OpenAPIUIBundlesUpdater.java [hunk 3]
```diff
@@ -32,7 +32,6 @@
 import org.osgi.framework.BundleContext;
 import org.osgi.framework.BundleException;
 import org.osgi.framework.FrameworkUtil;
-import org.osgi.framework.ServiceReference;
```
Comment (tjwatson) [inline dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/impl/OpenAPIUIBundlesUpdater.java:37]: The header here looks to still be using EPL-1.0.  It looks like you need to rebase your PR branch onto the latest `release` branch to be sure you are basing on the latest license headers.

Comment (tjwatson) [reply dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/impl/OpenAPIUIBundlesUpdater.java:37]: Also, update the copyright date once you rebase for the files you modified.

**Comment by KeenanMcCarthy** [comment]
#libby

**Comment by KeenanMcCarthy** [comment]
#build

**Review by tjwatson — APPROVED** [review]

**Pull request merged**